dandischema/consts.py

```diff
@@ -12,15 +12,16 @@
     "0.6.6",
     "0.6.7",
     "0.6.8",
+    DANDI_SCHEMA_VERSION,
 ]
 
 # ATM we allow only for a single target version which is current
 # migrate has a guard now for this since it cannot migrate to anything but current
 # version
 ALLOWED_TARGET_SCHEMAS = [DANDI_SCHEMA_VERSION]
+
 # This allows multiple schemas for validation, whereas target schemas focus on
 # migration.
-ALLOWED_VALIDATION_SCHEMAS = ALLOWED_TARGET_SCHEMAS + ALLOWED_INPUT_SCHEMAS
-
-if DANDI_SCHEMA_VERSION not in ALLOWED_INPUT_SCHEMAS:
-    ALLOWED_INPUT_SCHEMAS.append(DANDI_SCHEMA_VERSION)
+ALLOWED_VALIDATION_SCHEMAS = sorted(
+    set(ALLOWED_INPUT_SCHEMAS).union(ALLOWED_TARGET_SCHEMAS)
+)
```

Comment:
If `set(ALLOWED_TARGET_SCHEMAS)` is always a subset of `set(ALLOWED_INPUT_SCHEMAS)`, as it currently is, we can further simplify the definition of `ALLOWED_VALIDATION_SCHEMAS` to the following.

```suggestion
ALLOWED_VALIDATION_SCHEMAS = ALLOWED_INPUT_SCHEMAS
```